searxng/searxng
# Upgrade installation scripts and documentation

A preview of the documentation about the upgrade is available from here:

https://return42.github.io/searxng/admin/update-searxng.html#migrate-and-stay-tuned


**HINT:  last patch on top needs to be removed before this PR is merged!** 

    [brand] branch: searxng-install https://github.com/return42/searxng

This brand is needed to test the installation procedures as long this PR is not merged into master.

----

## What does this PR do?

```
- [utils/searxng.sh] implement new script to install SearXNG
- [doc] update documentation of the installation procedures
- [clean up] drop obsolete searx, filtron and morty install scripts
- [docs] add section "Migrate and stay tuned!"
```

## Why is this change important?

The infrastructure of SearXNG has changed dramatically over the past year. The installation scripts and documentation have not been adapted so far.

## How to test this PR locally?

----

First read the admin documentation, to see what have been changed ..

    make docs.live

----

Test new `./utils/searxng.sh` script ..

in LXCs, first build containers and install SearXNG by `install all`:

    sudo -H ./utils/lxc.sh build
    sudo -H ./utils/lxc.sh cmd -- FORCE_TIMEOUT=0 ./utils/searxng.sh install all
    
Choose to `install nginx`:

    sudo -H ./utils/lxc.sh cmd -- FORCE_TIMEOUT=0 ./utils/searxng.sh install nginx

or `install apache` HTTP server:

    sudo -H ./utils/lxc.sh cmd -- FORCE_TIMEOUT=0 ./utils/searxng.sh install apache
     
Run `instance check`:

    sudo -H ./utils/lxc.sh cmd -- FORCE_TIMEOUT=0 ./utils/searxng.sh instance check

Now you can test the SearXNG instances from the conatiners, to get the URLs of the containers use:

    sudo -H ./utils/lxc.sh show suite | grep SEARXNG_URL
    ...
    [searxng-ubu2004]    SEARXNG_URL          : http://10.146.148.203/searxng
    [searxng-ubu2110]    SEARXNG_URL          : http://10.146.148.219/searxng
    [searxng-fedora35]    SEARXNG_URL          : http://10.146.148.64/searxng
    [searxng-archlinux]    SEARXNG_URL          : http://10.146.148.135/searxng

----

Settings see https://github.com/return42/searxng/blob/searxng-install/utils/templates/etc/searxng/settings.yml#L3-L22

Features:

- the limiter should work (except on fedora35, see "Pitfalls of the Tyrant mode" in the uWSGI admn-doc)
- image_proxy should work on all instances
- cache busting (aka `static_use_hash`) should work

----

closes: https://github.com/searxng/searxng/issues/1112
